### ISSUE TITLE
How does this model get facial keypoint coordinates?

### ISSUE THREAD

**Issue by chuckle-lee** [google/mediapipe#3185]
Hi, I was looking at the papers and I feel confused. Seems it only tells us how the anchor is calculated (by modifying SDD), but it didn't mention how the 6 facial keypoints are calculated. Does anybody have ideas about it? Thanks a lot!

**Comment by sureshdagooglecom** [comment]
Hi @chuckle-lee ,
Mediapipe's landmarks value is normalized by the width and height of the image. After, getting the landmark value simply multiple the x of the landmark with the width of your image and y of the landmark with the height of your image and below is the code snippet to get co-ordinates.
 #in the cycle of capture
        if results.multi_face_landmarks:
            coord= results.multi_face_landmarks.landmark[0]
            print(''.join(['(',str(coord.x),',',str(coord.y),',',str(coord.z)  ,')']))

**Comment by chuckle-lee** [comment]
Hi @sureshdagooglecom, thanks for your reply! 
i am actually confused with the model design. Is it that after the feature extraction (5 blazeblock and 6 double blazeblock), the feature map is directly used for facial keypoint landmark estimation, which is done by convolution and regression?
thanks for your help!

**Comment by sureshdagooglecom** [comment]
Could you please  elaborate https://github.com/google/mediapipe/issues/3185#issuecomment-1073521286 this with more details.

**Comment by chuckle-lee** [comment]
@sureshdagooglecom 
hi, i want to know more about network design. The paper talks all about how the anchor is estimated and how they do training. I want to know how the network is designed to estimate the facial key points. 

From the paper, I know that the feature extraction part of the networks is composed of 5 blazeblocks and 6 double blazeblocks. Then they use SDD method to get the anchor. Is it that, after the feature extraction, a convolutional neural network is performed, which give the estimated coordinates of the facial keypoints? And then, is it that, the normalized difference between the estimated coordinates and the ground-truth is calculated for the purpose of training?

thanks for your help!

**Comment by sureshdagooglecom** [comment]
@chuckle-lee ,
Please check the github face detection [solutions] (https://github.com/google/mediapipe/blob/master/mediapipe/python/solutions/face_detection.py#:~:text=def%20get_key_point(,LEFT_EAR_TRAGION%20%3D%205) this is how facial key points will be calculated.

**Comment by chuckle-lee** [comment]
@sureshdagooglecom thanks!
